### PR TITLE
Fix the return values type

### PR DESCRIPTION
**Pull request by swapnili**
Fix the functions which returns negative values but the return type is unsigned. Also return valid errno instead of return -1 in case of errors.

### CODE REVIEW

File: kmod/muser.c [hunk 1]
```diff
@@ -1119,10 +1119,10 @@ static unsigned int get_minsz(unsigned int cmd)
 	case VFIO_DEVICE_SET_IRQS:
 		return offsetofend(struct vfio_irq_set, count);
 	}
-	return -1;
+	return -EOPNOTSUPP;
```
Comment (tmakatos) [inline kmod/muser.c]: `EINVAL` would be more suitable.

File: kmod/muser.c [hunk 2]
```diff
@@ -1135,20 +1135,19 @@ static unsigned int get_argsz(unsigned int cmd, struct mudev_cmd *mucmd)
 		return mucmd->muser_cmd.ioctl.data.irq_set.argsz;
 	}
 
-	return -1;
+	return -EOPNOTSUPP;
```
Comment (tmakatos) [inline kmod/muser.c]: ditto

File: kmod/muser.c [hunk 3]
```diff
@@ -1480,23 +1479,25 @@ static inline int mmap_done(struct mudev_cmd * const mucmd)
...
 		mucmd->pg_map.nr_pages = 0;
 	}
+
 	return ret;
 }
 
-static long libmuser_unl_ioctl(struct file *filep,
-				     unsigned int cmd, unsigned long arg)
+static long libmuser_unl_ioctl(struct file *filep, unsigned int cmd,
+			       unsigned long arg)
```
Comment (tmakatos) [inline kmod/muser.c]: white space

File: kmod/muser.c [hunk 4]
```diff
@@ -1552,20 +1549,20 @@ static long libmuser_unl_ioctl(struct file *filep, unsigned int cmd,
 		if (ret)
 			goto out;
 
+		mucmd_err = mucmd->muser_cmd.err;
```
Comment (tmakatos) [inline kmod/muser.c]: Is this just for readability?

File: kmod/muser.c (hunk 1 above)
Comment (swapnili) [reply kmod/muser.c]: The invalid(EINVAL) cmds are already checked at vfio layer, vfio only sends us the valid commands. It is at our layer either we support them or not, that is the reason EOPNOTSUPP looks more better.

File: kmod/muser.c (hunk 2 above)
Comment (swapnili) [reply kmod/muser.c]: ditto

File: kmod/muser.c (hunk 1 above)
Comment (tmakatos) [reply kmod/muser.c]: OK, I think it's worth a comment in our code that VFIO does this.

File: kmod/muser.c (hunk 3 above)
Comment (swapnili) [reply kmod/muser.c]: Its strange I do not see a white space in the file. Although the line is been adjusted to be started after the '('.
like: 
static long libmuser_unl_ioctl(struct file *filep, unsigned int cmd,
                                                  unsigned long arg)

File: kmod/muser.c (hunk 4 above)
Comment (swapnili) [reply kmod/muser.c]: yes

**Pull request merged**